src/pages/Home.jsx

```diff
@@ -1,12 +1,137 @@
-import React from "react";
+
+import React, { useState, useEffect } from "react";
+import Country from "../components/Country";
+import Modal from "../components/Modal";
 
 const Home = () => {
-    return (
-        // TODO: Home page
-        // Render Country component (components/Country.jsx) for each country
-        // Take data from (assets/CountriesData.json)
-        <div>Home</div>
-    );
-};
+  const [countries, setCountries] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedCountry, setSelectedCountry] = useState(null);
+
+  useEffect(() => {
+    const fetchCountries = async () => {
+      try {
+        const response = await fetch("https://restcountries.com/v3.1/all");
+        const data = await response.json();
```

Comment:
Save the hardcoded string into constants, and the fetch function should be extracted to a service that handles api requests